osu.Game.Rulesets.Osu/Scoring/OsuLegacyHealthProcessor.cs

```diff
@@ -82,6 +82,10 @@ protected override double GetHealthIncreaseFor(HitObject hitObject, HitResult re
                 case HitResult.LargeBonus:
                     increase = 0.01;
                     break;
+
+                case HitResult.HealthBonus:
+                    increase = 0.05;
```

Comment:
Why is this 0.05? This implementation is supposed to be matching legacy hp exactly.

In fact it appears likely that `HealthBonus` should not be used here at all and something was not updated properly because bonus ticks are explicitly excluded from enumeration above:

https://github.com/ppy/osu/blob/16c7c14602c4575c302e3495587c5b7f23369e4d/osu.Game.Rulesets.Osu/Scoring/OsuLegacyHealthProcessor.cs#L32-L34

so presumably the health bonus ticks should also be.

Reply:
I added the health bonus ticks to the above enumeration, and adjusted the HealthIncrease to 0.03.

I cross-referenced stable - in `osu.GameplayElements.HitObjects.Osu.SpinnerOsu` when "old algorithm" is not enabled (just replay stuff) and `if (this.scoringRotationCount > this.get_fullScoreRequiement())` returns an IncreaseScoreType that sets the health increase in `osu.GameModes.Play.Rulesets.Ruleset.IncreaseScoreHit` to `this.HpMultiplierNormal * 6.0` which I believe should correspond to the 0.03 in the code above.

The reason HealthBonus ticks need to be used here is due to the fact that stable does scoring differently - it has no bonus cap meaning you can gain these ticks until the end of the spinner object.